odl/solvers/advanced/proximal_operators.py

```diff
@@ -258,90 +470,242 @@ def __init__(self, sigma):
+    proximal_l2_squared : proximal without convex conjugate
+    """
+    lam = float(lam)
+
+    if g is not None and g not in space:
+        raise TypeError('{!r} is not an element of {!r}'.format(g, space))
+
+    class ProximalConvConjL2Squared(Operator):
+
+        """Proximal operator of the convex conjugate of the squared l2-norm."""
+
+        def __init__(self, sigma):
+            """Initialize the proximal operator.
+
+            Parameters
+            ----------
+            sigma : positive `float`
+                Step size parameter
+            """
+            self.sigma = float(sigma)
+            super().__init__(domain=space, range=space, linear=g is None)
+
+        def _call(self, x, out):
             """Apply the operator to ``x`` and stores the result in
             ``out``"""
 
             # (x - sig*g) / (1 + sig/lam)
 
             sig = self.sigma
-            out.lincomb(1 / (1 + sig / lam), x, -sig / (1 + sig / lam), g)
+            if g is None:
+                out.lincomb(1 / (1 + sig / lam), x)
+            else:
+                out.lincomb(1 / (1 + sig / lam), x, -sig / (1 + sig / lam), g)
 
-    return _ProximalConvConjL2
+    return ProximalConvConjL2Squared
 
 
-def proximal_convexconjugate_l1(space, lam=1, g=None):
-    """Proximal operator factory of the convex conjugate of the l1-semi-norm.
+def proximal_l2_squared(space, lam=1, g=None):
+    """Proximal operator factory of the squared l2-norm.
 
     Function for the proximal operator of the convex conjugate of the
-    functional F where F is an l1-semi-norm
+    functional F where F is the l2-norm
 
-        F(x) = lam || ||x-g||_p ||_1
+        F(x) =  lam 1/2 ||x - g||_2^2
 
-    with x and g elements in ``space``, scaling factor lam, and point-wise
-    magnitude ||x||_p of x. If x is vector-valued, ||x||_p is the point-wise
-    l2-norm across the vector components.
+    with x and g elements in ``space``, scaling factor lam, and given data g.
+
+    The proximal operator of ``sigma * F^*`` where ``sigma`` is a step size
+    is given by
+
+        prox[sigma * F^*](y) = (y - sigma g) / (1 + sigma/lam)
+
+    Parameters
+    ----------
+    space : `LinearSpace`
+        Domain of F(x). Needs to be a hilbert space (have an inner product)
+    g : ``space`` element
+        An element in ``space``
+    lam : positive `float`
+        Scaling factor or regularization parameter
+
+    Returns
+    -------
+    prox_factory : `callable`
+        Factory for the proximal operator to be initialized
+
+    See Also
+    --------
+    proximal_l2 : proximal without square
+    proximal_cconj_l2_squared : proximal for convex conjugate
+    """
 
-    The convex conjugate, F_cc, of F is given by the indicator function of
+    # TODO: optimize
+    prox_cc_l2_squared = proximal_cconj_l2_squared(space, lam=lam, g=g)
+    return proximal_cconj(prox_cc_l2_squared)
+
+
+def proximal_cconj_l1(space, lam=1, g=None, isotropic=False):
```

Comment:
As said above, correct but leaving out any mention of norm or dist in the function namw faciliates the dual interpretation as both norm and dist, depending on g.

If a useruses this without providing g, which is the usual case, then the word dist would only cause confusion.